var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -0,0 +1,65 @@
+##############################################################################
+# Copyright (c) 2017, Los Alamos National Security, LLC
+# Produced at the Los Alamos National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Legion(CMakePackage):
+    """Legion is a data-centric parallel programming system for writing portable high
+       performance programs targeted at distributed heterogeneous architectures.
+       Legion presents abstractions which allow programmers to describe properties of
+       program data (e.g. independence, locality). By making the Legion programming
+       system aware of the structure of program data, it can automate many of the tedious
+       tasks programmers currently face, including correctly extracting task- and
+       data-level parallelism and moving data around complex memory hierarchies. A novel
+       mapping interface provides explicit programmer controlled placement of data in the
+       memory hierarchy and assignment of tasks to processors in a way that is orthogonal
+       to correctness, thereby enabling easy porting and tuning of Legion applications to
+       new architectures.
+    """
+    homepage = "http://legion.stanford.edu/"
+    url      = "https://github.com/StanfordLegion/legion/tarball/legion-17.02.0"
+
+    version('develop', git='https://github.com/StanfordLegion/legion', branch='master')
+    version('17.02.0', '31ac3004e2fb0996764362d2b6f6844a')
+
+    variant('debug', default=False, description='Build debug version')
+
+    depends_on("cmake@3.1:", type='build')
+    depends_on("gasnet")
+
+    def build_type(self):
+        spec = self.spec
+        if '+debug' in spec:
+            return 'Debug'
+        else:
+            return 'Release'
+
+    def cmake_args(self):
+        return ['-DLegion_USE_GASNet=ON', '-DLegion_BUILD_EXAMPLES=ON']
+
+    def build(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make()
```

Comment:
Why is this method necessary? It seems to me it does the same thing as the default one. Am I missing something?